pkg/sqlutils/transactions.go

```diff
@@ -2,40 +2,57 @@
 package sqlutils
 
 import (
+	"context"
 	"fmt"
 
 	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
 )
 
 // WithTransactionRet is a wrapper function which handles creating, committing or rolling back a transaction
 // If there are any errors when executing the txFunc, the tx is rolled back. Otherwise, the tx is committed.
-func WithTransactionRet[T any](txPrep TransactionPreparer, txFunc func(*sqlx.Tx) (*T, error)) (*T, error) {
+func WithTransactionRet[T any](ctx context.Context, txPrep TransactionPreparer, txFunc func(*sqlx.Tx) (T, error)) (T, error) {
+	var (
+		result T
+		logger = appcontext.ZLogger(ctx)
+	)
+
 	tx, err := txPrep.Beginx()
 	if err != nil {
-		return nil, fmt.Errorf("error creating transaction %w", err)
+		return result, fmt.Errorf("error creating transaction %w", err)
 	}
 
-	result, errFunc := txFunc(tx)
-	if errFunc != nil {
-		rollbackErr := tx.Rollback()
-		if rollbackErr != nil {
-			return nil, fmt.Errorf("issue rolling back transaction: %w", rollbackErr)
+	defer func() {
```

Comment:
rollback in any scenario that would not result in `commit` running, such as `panic`

Reply:
we can definitely discuss this layout - it can also look like this:

```go
defer func() {
    if err := tx.Rollback(); err != nil {
        // handle err
    }
}()
```

the goal is to have `tx.Rollback()` always try to run should anything happen that prevents `tx.Commit()` from being tried

(running `tx.Rollback()` after `tx.Commit()` runs is a no-op but would return an error stating the `tx` is done, so having it in this `if/else` block just avoids that error)